src/xdp/trn_transit_xdp.c

```diff
@@ -175,44 +176,49 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 		flow->dport = pkt->inner_udp->dest;
 	}
 
-	/* Modify inner EtherHdr */
-	trn_set_dst_mac(pkt->inner_eth, ep->mac);
-
-	/* Keep overlay header, update outer header destinations */
-	trn_set_src_dst_ip_csum(pkt->ip, pkt->ip->saddr, ep->hip, pkt->data_end);
-	trn_set_dst_mac(pkt->eth, ep->hmac);
-
 	/* Generate Direct Path request */
-	pkt->fctx.opcode = XDP_FLOW_OP_ENCAP;
+	pkt->fctx.opcode = bpf_htonl(XDP_FLOW_OP_ENCAP);
 	pkt->fctx.opdata.encap.dip = epkey.ip;
 	pkt->fctx.opdata.encap.dhip = ep->hip;
 	trn_set_mac(pkt->fctx.opdata.encap.dmac, ep->mac);
 	trn_set_mac(pkt->fctx.opdata.encap.dhmac, ep->hmac);
-	pkt->fctx.opdata.encap.timeout = TRAN_DP_FLOW_TIMEOUT;
+	pkt->fctx.opdata.encap.timeout = bpf_htons(TRAN_DP_FLOW_TIMEOUT);
```

Comment:
Ditto, here. (Tradeoff of readability vs performance since this is in seconds) What do you think?

Reply:
this is actually a bug, should be byte swapped: bpf_htons(TRAN_DP_FLOW_TIMEOUT)
And this won't cause performance impact as TRAN_DP_FLOW_TIMEOUT is const, the bpf_htons(TRAN_DP_FLOW_TIMEOUT) will actually been done at compile time, not runtime :), but it's good we start to think performance impact in details.

Reply:
Strange, In my local repo, I was using bpf_htons(TRAN_DP_FLOW_TIMEOUT);

Reply:
Holdon, I mis-read the diff. I thought I didn't change to bigendian. The diff is correct. This is different from changing to network order for some bpfmap fields. Those are variables which will be used either for comparing with packet buffer data, or fill in packet buffer, so change to network order at update time avoided runtime byte swapping per packet in XDP. This one we byteswap const value doesn't affect runtime as the swap is done at compile time for const, won't affect runtime at all.